azurelinuxagent/common/osutil/default.py

```diff
@@ -167,11 +198,13 @@ def _delete_rule(self, rule):
         code is non-zero or the limit has been reached.
         """
         for i in range(1, 100):
-            rc = shellutil.run(rule, chk_err=False)
-            if rc == 1:
-                return
-            elif rc == 2:
-                raise Exception("invalid firewall deletion rule '{0}'".format(rule))
+            try:
+                rc = shellutil.run_command(rule)
+            except CommandError as e:
+                if e.returncode == 1:
+                    return
```

Comment:
Why is a returncode of 1 not an error? Should we document this as a comment?

Reply:
I image the code was written like this to avoid flagging an error if the rule does not exist, though this is not very robust.
I was tempted to change the logic, but at the end decided against it. I am not sure about the intention of the loop either.